foxpuppet/windows.py

```diff
@@ -2,44 +2,50 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
-from contextlib import contextmanager
-
-from selenium.webdriver.common.by import By
 from selenium.webdriver.support.wait import WebDriverWait
 
 
-class Windows(object):
+class WindowManager(object):
 
     def __init__(self, selenium):
         self.selenium = selenium
+        self.windows = None
+
+        self.populate_windows()
+
+    def populate_windows(self):
+        self.windows = []
+        for handle in self.selenium.window_handles:
+            self.windows.append(BaseWindow(self.selenium, handle))
+
+
+class BaseWindow(object):
```

Comment:
I suggested simply `Window`, but on reflection I'm happy with `BaseWindow`. This is also what I've done with the notifications patch. I'm also tempted to suggest that `BaseWindow` should be an abstract base class, and that it should not be instantiated directly.